### PR TITLE
OCM-7960 | ci: Automated id:38770,38778,72162,46187,45176,44511

### PR DESCRIPTION
**Pull request by yuwang-RH**
[OCM-7960](https://issues.redhat.com//browse/OCM-7960) | ci: Automated id:38770,38778,72162,46187,45176,44511

The logs see here: https://privatebin.corp.redhat.com/?386a22dc6d5c23f6#ADXBN2N9zdvRDXoyV1ngJ48odSvjLtGqyRAsdy2oCEJs

### CODE REVIEW

File: tests/e2e/test_rosacli_cluster.go [hunk 1]
```diff
@@ -297,4 +297,216 @@ var _ = Describe("Edit cluster",
 				Expect(err).To(BeNil())
 				Expect(stdout.String()).To(ContainSubstring(fmt.Sprintf("Creating cluster '%s' should succeed", clusterName)))
 			})
+		It("Check the basic validation for the classic rosa cluster creation by the rosa cli - [id:38770]", labels.High, labels.NonHCPCluster, func() {
```
Comment (xueli181114) [inline tests/e2e/test_rosacli_cluster.go:300]: Is it a day1 negative case? Define a new label with day1-validation and label these kind of cases.

Comment (xueli181114) [reply tests/e2e/test_rosacli_cluster.go:300]: Validation cases maybe in importance of `Medium`?

File: tests/e2e/test_rosacli_cluster.go [hunk 2]
```diff
@@ -297,4 +297,216 @@ var _ = Describe("Edit cluster",
...
+				"2":  "Multi AZ cluster requires at least 3 compute nodes",
+				"0":  "Multi AZ cluster requires at least 3 compute nodes",
+				"-3": "must be non-negative",
+				"5":  "Multi AZ clusters require that the number of compute nodes be a multiple of 3",
+			}
+			invalidReplicasTypeErrorMap := map[string]string{
+				"4.5":  "invalid argument \"4.5\" for \"--replicas\" flag",
+				"five": "invalid argument \"five\" for \"--replicas\" flag",
+			}
+			rosalCommand, err := config.RetrieveClusterCreationCommand(ciConfig.Test.CreateCommandFile)
```
Comment (xueli181114) [inline tests/e2e/test_rosacli_cluster.go:324]: If this is a day1-negative case, we need a discussion about the day1 negative cases. Combine it to current CI job or have another job. If we pick the second resolution, then there won't be creation command file existing.

File: tests/e2e/test_rosacli_cluster.go (hunk 1 above)
Comment (yuwang-RH) [reply tests/e2e/test_rosacli_cluster.go:300]: @xueli181114 Yes. It is a day-1 validation testing, however it needs the command of creating cluster which generate in job day-1 provision. If add such "day1-validation", will it be selected without day-q provision in future?

File: tests/e2e/test_rosacli_cluster.go [hunk 3]
```diff
@@ -297,4 +297,216 @@ var _ = Describe("Edit cluster",
...
+			Expect(err).To(BeNil())
+			rosalCommand.AddFlags("--dry-run")
+			originalClusterName := rosalCommand.GetFlagValue("--cluster-name", true)
+			originalMachineType := rosalCommand.GetFlagValue("--compute-machine-type", true)
+			originalRegion := rosalCommand.GetFlagValue("--region", true)
+			originalReplicas := rosalCommand.GetFlagValue("--replicas", true)
+
+			for _, cn := range invalidClusterNames {
```
Comment (xueli181114) [inline tests/e2e/test_rosacli_cluster.go:332]: Let's put related variable definition together with the steps.

File: tests/e2e/test_rosacli_cluster.go [hunk 4]
```diff
@@ -297,4 +297,216 @@ var _ = Describe("Edit cluster",
...
+			By("Check the validation for billing-account for classic sts cluster")
+			rosalCommand.ReplaceFlagValue(map[string]string{
+				"--region": originalRegion,
+			})
+			rosalCommand.AddFlags("--billing-account", "123456789")
+			stdout, err = rosaClient.Runner.RunCMD(strings.Split(rosalCommand.GetFullCommand(), " "))
+			Expect(err).NotTo(BeNil())
+			Expect(stdout.String()).To(ContainSubstring("Billing accounts are only supported for Hosted Control Plane clusters"))
+		})
+		It("Validation for deletion ROSA cluster will work via rosacli	- [id:38778]", labels.High, func() {
```
Comment (xueli181114) [inline tests/e2e/test_rosacli_cluster.go:399]: Put validation to importance `Medium`. And day3 label needed.

File: tests/e2e/test_rosacli_cluster.go [hunk 5]
```diff
@@ -297,4 +297,216 @@ var _ = Describe("Edit cluster",
...
+			Expect(err).NotTo(BeNil())
+			Expect(out.String()).To(ContainSubstring("There is no cluster with identifier or name"))
+
+			By("Delete with unknown flag --interactive")
+			out, err = clusterService.Delete(notExistID, "-y", "--interactive")
+			Expect(err).NotTo(BeNil())
+			Expect(out.String()).To(ContainSubstring("unknown flag: --interactive"))
+		})
+
+		It("Validation for deletion ROSA cluster will work via rosacli	- [id:72162]", labels.High, func() {
```
Comment (xueli181114) [inline tests/e2e/test_rosacli_cluster.go:419]: Title is wrong.

File: tests/e2e/test_rosacli_iam_roles.go [hunk 6]
```diff
@@ -1049,4 +1073,50 @@ var _ = Describe("Edit IAM",
 					Expect(rosaClient.Parser.TextData.Input(output).Parse().Tip()).To(ContainSubstring("OIDC provider already exists"))
 				}
 			})
+
+		It("List account-roles by rosa-cli - [id:44511]", func() {
```
Comment (xueli181114) [inline tests/e2e/test_rosacli_iam_roles.go:1077]: Just want to double confirm about the label. `day2`? `High`?

**Comment by yuwang-RH** [comment]
@xueli181114 Could you help take a look again? I have updated based on all your comments.

File: tests/e2e/test_rosacli_cluster.go [hunk 7]
```diff
@@ -2,16 +2,21 @@ package e2e
...
 
 	. "github.com/onsi/ginkgo/v2"
 	. "github.com/onsi/gomega"
 
 	ciConfig "github.com/openshift/rosa/tests/ci/config"
 	"github.com/openshift/rosa/tests/ci/labels"
+	con "github.com/openshift/rosa/tests/utils/common/constants"
 	"github.com/openshift/rosa/tests/utils/config"
 	"github.com/openshift/rosa/tests/utils/exec/rosacli"
+	profileHanddler "github.com/openshift/rosa/tests/utils/profilehandler"
```
Comment (yasun1) [inline tests/e2e/test_rosacli_cluster.go:19]: It is meaningless for us to define the alias as same as the package name. Maybe `profile` could be better.

File: tests/e2e/test_rosacli_cluster.go [hunk 8]
```diff
@@ -303,3 +308,230 @@ var _ = Describe("Edit cluster",
...
+			rosalCommand.ReplaceFlagValue(map[string]string{
+				"--compute-machine-type": invalidMachineType,
+				"--cluster-name":         originalClusterName,
+			})
+			stdout, err := rosaClient.Runner.RunCMD(strings.Split(rosalCommand.GetFullCommand(), " "))
+			Expect(err).NotTo(BeNil())
+			Expect(stdout.String()).To(ContainSubstring("Expected a valid machine type"))
+
+			By("Check the validation for replicas")
+
```
Comment (yasun1) [inline tests/e2e/test_rosacli_cluster.go:389]: A redaudant space line.

File: tests/e2e/test_rosacli_cluster.go [hunk 9]
```diff
@@ -303,3 +308,230 @@ var _ = Describe("Edit cluster",
...
+			profile.NamePrefix = con.DefaultNamePrefix
+
+		})
+
+		AfterEach(func() {
+			errs := profileHanddler.DestroyResourceByProfile(profile, rosaClient)
+			Expect(len(errs)).To(Equal(0))
+		})
+
+		It("To check the basic validation for the classic rosa cluster creation by the rosa cli - [id:38770]", labels.Medium, labels.Day1Validation, labels.NonHCPCluster, func() {
```
Comment (yasun1) [inline tests/e2e/test_rosacli_cluster.go:344]: lower case for `To`

File: tests/e2e/test_rosacli_cluster.go [hunk 10]
```diff
@@ -303,3 +308,230 @@ var _ = Describe("Edit cluster",
...
+			accrolePrefix := "testAr45176"
+			path := "/a/b/"
+			ocmResourceService := rosaClient.OCMResource
+			clusterName := "cluster45176"
+			operatorRolePrefix := "cluster45176-xvfa"
+			By("Create account-roles in low version 4.14")
+			output, err := ocmResourceService.CreateAccountRole("--mode", "auto",
+				"--prefix", accrolePrefix,
+				"--path", path,
+				"--version", "4.14",
```
Comment (yasun1) [inline tests/e2e/test_rosacli_cluster.go:465]: Hardcoding the version means we need to update it in the future. It is better to get the lower version dynamically.

File: tests/e2e/test_rosacli_iam_roles.go [hunk 11]
```diff
@@ -1137,3 +1161,65 @@ var _ = Describe("Edit IAM",
...
+		BeforeEach(func() {
+			By("Init the client")
+			rosaClient = rosacli.NewClient()
+			ocmResourceService = rosaClient.OCMResource
+		})
+
+		It("to list account-roles by rosa-cli - [id:44511]", labels.High, func() {
+			accrolePrefix := "arPrefix44511"
+			path := "/a/b/"
+			version := "4.15"
```
Comment (yasun1) [inline tests/e2e/test_rosacli_iam_roles.go:1184]: same recommend as above on version.

**Pull request merged**